UKHomeOffice/nmsw-ui
# NMSW-440 Add error mapping and tests for value too long errors

# Ticket

NMSW-440

----

## AC

Given I have uploaded a FAL 1 file
When there are fields with character limits
And my value excedes that character limit
Then I am shown an error message to help me fix them


Cell | Field | Rule | Error Message
-- | -- | -- | --
B3 | nameofship | max 35 characters | Enter the name of the ship using 35 characters or less
F3 | callsign | max 35 characters | Enter the call sign using 35 characters or less
B4 | signatory | max 35 characters | Enter the name of the master using 35 characters or less
D4 | flagstate | max 3 letters, or more if the excess characters are spaces | Enter the code for the country using 3 letters as given in the ISO list of 3166-1 alpha 3 codes
B13 | cargo | max 35 characters | Enter the cargo details using 35 characters or less



----

## To test

- upload a spreadsheet that has values that excede the max character lengths for the fields in the table above
- submit the file
- > you should see the error wording as the `Error message` from the table above
- > the order should match the screenshot on the Jira ticket

----

## Notes
